OutOfSchool/OutOfSchool.WebApi.Tests/Controllers/AchievementControllerTest.cs

```diff
@@ -31,6 +36,139 @@ public void Setup()
         controller = new AchievementController(achievementService.Object, providerService.Object, providerAdminService.Object, workshopService.Object);
     }
 
+    [Test]
+    public async Task CreateAchievement_WhenAchievementDtoIsNull_ShouldReturnBadRequest()
+    {
+        // Act
+        var result = await controller.Create(null).ConfigureAwait(false);
+
+        // Assert
+        Assert.IsInstanceOf<BadRequestObjectResult>(result);
+    }
+
+    [Test]
+    public async Task CreateAchievement_WhenWorkshopIdIsNotValid_ShouldReturnNotFound()
+    {
+        // Arrange
+        var dto = new AchievementCreateDTO();
+        workshopService.Setup(s => s.Exists(It.IsAny<Guid>())).ReturnsAsync(false);
+
+        // Act
+        var result = await controller.Create(dto).ConfigureAwait(false);
+
+        // Assert
+        Assert.IsInstanceOf<NotFoundObjectResult>(result);
+    }
+
+    [Test]
+    public async Task CreateAchievement_WhenProviderIsBlocked_ShouldReturnForbidden()
+    {
+        // Arrange
+        var dto = new AchievementCreateDTO();
+        workshopService.Setup(s => s.Exists(It.IsAny<Guid>())).ReturnsAsync(true);
+        providerService.Setup(s => s.GetProviderIdForWorkshopById(It.IsAny<Guid>())).ReturnsAsync(Guid.NewGuid());
+        providerService.Setup(s => s.IsBlocked(It.IsAny<Guid>())).ReturnsAsync(true);
+
+        // Act
+        var result = await controller.Create(dto).ConfigureAwait(false) as ObjectResult;
+
+        // Assert
+        Assert.That(result, Is.Not.Null);
+        Assert.AreEqual(403, result.StatusCode);
+    }
+
+    [Test]
+    public async Task CreateAchievement_UserDontHaveRights_ShouldReturnForbidden()
+    {
+        // Arrange
+        var dto = new AchievementCreateDTO();
+        workshopService.Setup(s => s.Exists(It.IsAny<Guid>())).ReturnsAsync(true);
+        providerService.Setup(s => s.GetProviderIdForWorkshopById(It.IsAny<Guid>())).ReturnsAsync(Guid.NewGuid());
+        providerService.Setup(s => s.IsBlocked(It.IsAny<Guid>())).ReturnsAsync(false);
+
+        var user = new ClaimsPrincipal(new ClaimsIdentity(
+            new Claim[]
+            {
+                new Claim(ClaimTypes.Role, nameof(Role.Parent).ToLower()),
+                new Claim("subrole", nameof(Subrole.None).ToLower()),
+                new Claim("sub", Guid.NewGuid().ToString()),
+            }));
+
+        controller.ControllerContext.HttpContext = new DefaultHttpContext { User = user };
```

Comment:
(Test code readability) Selected part is duplicated in many tests. Please consider implementing and usage of 
`private void SetContextUser(Role role, Subrole subrole, Guid sub)`

Reply:
Added method